### ISSUE TITLE
Bug in nltk.draw.dispersion_plot with nltk 3.8.1, matplotlib-base 3.8.0, matplotlib-inline 0.1.6 and numpy 1.26

### ISSUE THREAD

**Issue by m-d-grunnill** [nltk/nltk#3206]
Following along the exercise in your workbook [chapter 1](https://www.nltk.org/book/ch01.html). 

I do not get the same plot when I use:
```
text4.dispersion_plot(["citizens", "democracy", "freedom", "duties", "America"])
```

After using the words singularly with the function, I believe 'citizens' output has been swapped with 'America' and 'democracy'  output has been swapped with duties.

**Comment by SarthakNikhal** [comment]
Can I work on this issue?

**Comment by rmalouf** [comment]
Is this already fixed by #3150?
